File: SOURCE/SmartParkingAPI/Models/Entities/Reservation.cs
using SmartParkingAPI.Models.Contracts;

namespace SmartParkingAPI.Models.Entities
{
    public class Reservation : IEntity
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int ParkingSlotId { get; set; }
        public DateTime ReservationTime { get; set; }
        public string? Status { get; set; } // "Confirmed", "Cancelled"

        public User? User { get; set; }
        public ParkingSlot? ParkingSlot { get; set; }
    }
}

File: SOURCE/smart_parking_app/lib/models/reservation.dart
class Reservation {
  final int id;
  final int parkingSlotId;
  final String status;
  final String reservationTime;

  Reservation({
    required this.id,
    required this.parkingSlotId,
    required this.status,
    required this.reservationTime,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      id: json['id'],
      parkingSlotId: json['parkingSlotId'],
      status: json['status'],
      reservationTime: json['reservationTime'],
    );
  }
}
